pkg/volume/aws_ebs/aws_util.go

```diff
@@ -214,6 +214,26 @@ func attachDiskAndVerify(b *awsElasticBlockStoreMounter, xvdBeforeSet sets.Strin
 		}
 	}
 
+	glog.V(5).Infof("Failed to attach volume %s: %v", b.volumeID, attachError)
+
+	if awsCloud != nil {
+		detached, err := awsCloud.IsDiskDetached(b.volumeID)
+		if err == nil && !detached {
```

Comment:
That's why I check for 'detached' status. All other states ('attached', 'attaching', 'detaching' and who knows what else) are considered bad and result in DetachDisk.call.